scripts/rocksdb-compatibility/rocksdb.py

```diff
@@ -0,0 +1,93 @@
+from cffi import FFI
+from contextlib import contextmanager
+
+ffi = FFI()
+
+ffi.cdef("""
+typedef struct rocksdb_t rocksdb_t;
+typedef struct rocksdb_options_t rocksdb_options_t;
+typedef struct rocksdb_readoptions_t rocksdb_readoptions_t;
+typedef struct rocksdb_iterator_t rocksdb_iterator_t;
+
+rocksdb_options_t* rocksdb_options_create(void);
+void rocksdb_options_destroy(rocksdb_options_t*);
+void rocksdb_options_set_create_if_missing(rocksdb_options_t*, unsigned char);
+
+rocksdb_t* rocksdb_open(const rocksdb_options_t* options, const char* name, char** errptr);
+void rocksdb_close(rocksdb_t* db);
+
+rocksdb_readoptions_t* rocksdb_readoptions_create(void);
+void rocksdb_readoptions_destroy(rocksdb_readoptions_t*);
+
+rocksdb_iterator_t* rocksdb_create_iterator(rocksdb_t* db, const rocksdb_readoptions_t* options);
+void rocksdb_iter_destroy(rocksdb_iterator_t* iter);
+void rocksdb_iter_seek_to_first(rocksdb_iterator_t* iter);
+unsigned char rocksdb_iter_valid(const rocksdb_iterator_t* iter);
+void rocksdb_iter_next(rocksdb_iterator_t* iter);
+const char* rocksdb_iter_key(const rocksdb_iterator_t* iter, size_t* klen);
+const char* rocksdb_iter_value(const rocksdb_iterator_t* iter, size_t* vlen);
+""")
+
+# Load the library
+rocksdb = ffi.dlopen("librocksdb.so")
+
+@contextmanager
+def rocksdb_options(create_if_missing=False):
+    opts = rocksdb.rocksdb_options_create()
+    rocksdb.rocksdb_options_set_create_if_missing(opts, int(create_if_missing))
+    try:
+        yield opts
+    finally:
+        rocksdb.rocksdb_options_destroy(opts)
+
+@contextmanager
+def open_db(path, options):
+    err_ptr = ffi.new("char**")
+    db = rocksdb.rocksdb_open(options, path.encode('utf-8'), err_ptr)
+    if err_ptr[0] != ffi.NULL:
+        raise RuntimeError("Open error: " + ffi.string(err_ptr[0]).decode())
+    try:
+        yield db
+    finally:
+        rocksdb.rocksdb_close(db)
+
+@contextmanager
+def read_iter(db):
```

Comment:
This should better be a generator, I think. Should leave as a follow-up.

BTW, the reason I'm hand-rolling a binding is because other existing libraries is not using dynamic linking.